mlflow_databricks_artifacts/store/entrypoint.py

```diff
@@ -0,0 +1,74 @@
+import json
+import os
+import posixpath
+
+from mlflow.exceptions import MlflowException
+
+try:
+    from mlflow.store.artifact.dbfs_artifact_repo import DbfsRestArtifactRepository
+    from mlflow.store.artifact.local_artifact_repo import LocalArtifactRepository
+except ImportError:
+    # Older versions of MLflow placed artifact repository implementations
+    # in the top-level `mlflow.store` module, rather than the
+    # `mlflow.store.artifact` module; `DbfsArtifactRepository` was also
+    # renamed to `DbfsRestArtifactRepository`
+    from mlflow.store.dbfs_artifact_repo import DbfsArtifactRepository as DbfsRestArtifactRepository
+    from mlflow.store.local_artifact_repo import LocalArtifactRepository
+
+
+def dbfs_artifact_repo_factory(artifact_uri):
```

Comment:
This is the entrypoint for the MLflow ACL'ed artifacts plugin. We override the artifact repository registry mapping for `dbfs` to point here.

One thing that this method does not currently do is check for the case where the version of MLflow is >= 1.9.1. In those cases, this plugin really isn't necessary, and we should fall back to the default MLflow implementation. As a strawman proposal, we can do this by checking for the presence of `mlflow/store/artifact/dbfs_artifact_repo.py#dbfs_artifact_repo_factory` in conjunction with the MLflow version. If this method is already defined in MLflow and the MLflow version is >= 1.9.1, we can instead fall back to calling this method.

As an aside, it might be useful to embed some notion of version-dependent plugins in MLflow going forward (e.g., I only want to use this plugin for MLflow versions up to x.y.z).